### PR TITLE
Fix bug for datetime in GHRSST Ioda Converter

### LINKED ISSUE #1019: GHRSST converter is not working properly
I suspect the datetime is wrong.
There's an example of processed obs here:
```bash
[hfe03 /scratch1/NCEPDEV/stmp2/Guillaume.Vernieres/runs/cp1]$ du -sh cp1.e/COMROOT/cp1.e/gdas.20210702/00/obs/gdas.t00z.sst*
12M	cp1.e/COMROOT/cp1.e/gdas.20210702/00/obs/gdas.t00z.sst_abi_g16_l3c.2021070200.nc4
9.7M	cp1.e/COMROOT/cp1.e/gdas.20210702/00/obs/gdas.t00z.sst_abi_g17_l3c.2021070200.nc4
12M	cp1.e/COMROOT/cp1.e/gdas.20210702/00/obs/gdas.t00z.sst_ahi_h08_l3c.2021070200.nc4
1.5M	cp1.e/COMROOT/cp1.e/gdas.20210702/00/obs/gdas.t00z.sst_avhrr_ma_l3u.2021070200.nc4
1.8M	cp1.e/COMROOT/cp1.e/gdas.20210702/00/obs/gdas.t00z.sst_avhrr_mb_l3u.2021070200.nc4
2.3M	cp1.e/COMROOT/cp1.e/gdas.20210702/00/obs/gdas.t00z.sst_viirs_n20_l3u.2021070200.nc4
2.3M	cp1.e/COMROOT/cp1.e/gdas.20210702/00/obs/gdas.t00z.sst_viirs_npp_l3u.2021070200.nc4
```
The datetime doesn't seem correct. The output file after DA are here:
```bash
[hfe03 /scratch1/NCEPDEV/stmp2/Guillaume.Vernieres/runs/cp1]$ du -sh /scratch1/NCEPDEV/stmp2/Guillaume.Vernieres/RUNDIRS/cp1.e/gdasocnanal_00/diags/*
39M	/scratch1/NCEPDEV/stmp2/Guillaume.Vernieres/RUNDIRS/cp1.e/gdasocnanal_00/diags/adt_rads_all.2021070200.nc4
24M	/scratch1/NCEPDEV/stmp2/Guillaume.Vernieres/RUNDIRS/cp1.e/gdasocnanal_00/diags/icec_amsr2_north.2021070200.nc4
48M	/scratch1/NCEPDEV/stmp2/Guillaume.Vernieres/RUNDIRS/cp1.e/gdasocnanal_00/diags/icec_amsr2_south.2021070200.nc4
2.3M	/scratch1/NCEPDEV/stmp2/Guillaume.Vernieres/RUNDIRS/cp1.e/gdasocnanal_00/diags/sst_abi_g16_l3c.2021070200.nc4
16K	/scratch1/NCEPDEV/stmp2/Guillaume.Vernieres/RUNDIRS/cp1.e/gdasocnanal_00/diags/sst_abi_g17_l3c.2021070200.nc4
16K	/scratch1/NCEPDEV/stmp2/Guillaume.Vernieres/RUNDIRS/cp1.e/gdasocnanal_00/diags/sst_ahi_h08_l3c.2021070200.nc4
60K	/scratch1/NCEPDEV/stmp2/Guillaume.Vernieres/RUNDIRS/cp1.e/gdasocnanal_00/diags/sst_avhrr_ma_l3u.2021070200.nc4
20K	/scratch1/NCEPDEV/stmp2/Guillaume.Vernieres/RUNDIRS/cp1.e/gdasocnanal_00/diags/sst_avhrr_mb_l3u.2021070200.nc4
20K	/scratch1/NCEPDEV/stmp2/Guillaume.Vernieres/RUNDIRS/cp1.e/gdasocnanal_00/diags/sst_viirs_n20_l3u.2021070200.nc4
64K	/scratch1/NCEPDEV/stmp2/Guillaume.Vernieres/RUNDIRS/cp1.e/gdasocnanal_00/diags/sst_viirs_npp_l3u.2021070200.nc4
```
The only non-empty output is ABI on GOES16.

Could you have a look at the `datetime` in the sst files here `/scratch1/NCEPDEV/stmp2/Guillaume.Vernieres/runs/cp1/cp1.e/COMROOT/cp1.e/gdas.20210702/00/obs/ @apchoiCMD ?

### PR DESCRIPTION
**Pull request by apchoiCMD**
- Resolve the FillValue issue within obs/metadata calculation loop
- Fix the encountering a precision issue when converting the integer values for datetime issue previously occurred
- Fixes #1019

### CODE REVIEW

File: utils/obsproc/Ghrsst2Ioda.h [hunk 1]
```diff
@@ -143,9 +145,13 @@ namespace gdasapp {
           // TODO(Somebody): add sampled std. dev. of sst to the total obs error
           obserror[i][j] = static_cast<float>(sstObsErr[index]) * errScaleFactor + errOffSet;
 
-          // epoch time in seconds
-          seconds[i][j]  = static_cast<float>(sstdTime[index]) * dtimeScaleFactor
-                         + static_cast<float>(refTime[0]);
+          // epoch time in seconds and avoid to use FillValue in calculation
+          if (sstdTime[index] == dtimeFillValue) {
+            seconds[i][j] = 0;
```
Comment (guillaumevernieres) [inline utils/obsproc/Ghrsst2Ioda.h:150]: how is storing 0 any better? Are these values not used when superobing?

Comment (AndrewEichmann-NOAA) [reply utils/obsproc/Ghrsst2Ioda.h:150]: I would expect that a time with the fill value would indicate an observation to be filtered out

Comment (apchoiCMD) [reply utils/obsproc/Ghrsst2Ioda.h:150]: Those are used and will be divided by counts,

https://github.com/NOAA-EMC/GDASApp/blob/401bd1e88c09934ada0a7f7184122ed280113d6d/utils/obsproc/superob.h#L54

At the end, those are filtered if less than zero
https://github.com/NOAA-EMC/GDASApp/blob/5a23b35ea9a2f0ed61a780c4cd2c08dbee876f44/utils/obsproc/Ghrsst2Ioda.h#L214

Am I thinking wrong?

Comment (guillaumevernieres) [reply utils/obsproc/Ghrsst2Ioda.h:150]: ha, ok. You mean those are NOT used.

Comment (guillaumevernieres) [reply utils/obsproc/Ghrsst2Ioda.h:150]: I'm still a bit paranoid @apchoiCMD , could you redo those histogram of time in the original and ioda files? Make sure we're not off again?

Comment (guillaumevernieres) [reply utils/obsproc/Ghrsst2Ioda.h:150]: > I would expect that a time with the fill value would indicate an observation to be filtered out

As @AndrewEichmann-NOAA is suggesting, why not filter out based on the fillvalue @apchoiCMD , as opposed to suing 0 ?

Comment (apchoiCMD) [reply utils/obsproc/Ghrsst2Ioda.h:150]: > I'm still a bit paranoid @apchoiCMD , could you redo those histogram of time in the original and ioda files? Make sure we're not off again?

Expected and wanted to do, I am already on it-

Comment (apchoiCMD) [reply utils/obsproc/Ghrsst2Ioda.h:150]: - 25 files of ABI_G16 in 20210702.00, Plot compares Raw and Ioda

16_Before_After

Comment (guillaumevernieres) [reply utils/obsproc/Ghrsst2Ioda.h:150]: Still does not look correct @apchoiCMD .

**Comment by guillaumevernieres** [comment]
I assume this was closed by mistake @apchoiCMD ?

**Comment by apchoiCMD** [comment]
> I assume this was closed by mistake @apchoiCMD ?

Oh, thanks- I don't even know it is closed.

**Review by guillaumevernieres — COMMENTED** [review]
Looks like you fixed the core of the issue, thanks @apchoiCMD ! ~But is the superobing still OK?~

**Review by guillaumevernieres — APPROVED** [review]
Good catch @apchoiCMD . I'm going to merge this, along with the last analysis that you shared, the stats from @KatherineLukens-NOAA 's runs seems to indicate that what you did is correct.

**Pull request merged**